ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,862 @@
+3. The workunit write activity will start its input.
+4. The index read activity will call the helper functions to get the filename, resolve the index, and create the filter.
+5. The workunit activity requests a row from its input.
+6. The index read finds the first row, and calls the helper's transform() method to create an output row.
+7. The workunit activity saves that row away (possibly using the serializer provided by the IOutputMetaData interface
+   implemented by the class mx1.
+8. Back to step 5 workunit write reading a row from its input, until end of file is returned (notified as two NULL
+   rows in a row).
+9. Workunit write commits the results and finishes.
+
+The execution generally switches back and forth between the code in the engines, and the members of the generated helper
+classes.
+
+There are some other details of query execution that are worth highlighting:
+
+Child Queries
+    Some activities perform complicated operations on child datasets of the input rows.  E.g., remove all duplicate
+    people who are marked as living at this address.  This will create a "child query" in the graph - i.e., a
+    nested graph within a subgraph which may be executed each time a new input row is processed by the containing
+    activity.
+    (The graphs of activities for the child graph are created at the same time as the rest of the activities, but 
+    they are reinitialised and re-executed for each input row processed by the parent activity.)
+
+Other helper functions
+    The generated code contains other functions which are used to describe the meta information for the rows
+    processed within the graph.
+    E.g., the following class describes the output from the index read activity::
+    
+        struct mi1 : public CFixedOutputMetaData {
+            inline mi1() : CFixedOutputMetaData(120) {}
+            virtual const RtlTypeInfo * queryTypeInfo() const { return &ty1; }
+        } mx1;
+        
+    This represents a fixed size row which occupies 120 bytes.  The object returned by the queryTypeInfo() function
+    provides information about the types of the fields::
+    
+        const RtlStringTypeInfo ty2(0x4,40);
+        const RtlFieldStrInfo rf1("name",NULL,&ty2);
+        const RtlStringTypeInfo ty3(0x4,80);
+        const RtlFieldStrInfo rf2("address",NULL,&ty3);
+        const RtlFieldInfo * const tl4[] = { &rf1,&rf2, 0 };
+        const RtlRecordTypeInfo ty1(0xd,120,tl4);
+
+    I.e. a string column, length 40 called "name", followed by a string column, length 80 called "address".  The
+    interface IOutputMetaData in eclhelper.hpp is key to understanding how the rows are processed.
+
+Inline dataset operations
+    The rule mentioned at the start - that the generated code doesn't contain any knowledge of how to perform a
+    particular dataset operation - does have one notable exception.  Some operations on child datasets are very
```

Comment:
- **"**that the ...operation**"** - 
Probably need to change doesn't -> does not (as may have been done in the source text).